infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -298,10 +305,25 @@ private void handleException(Throwable ex, ChannelHandlerContext ctx, RequestMsg
      */
     @RequestHandler(type = PayloadCase.WRITE_LOG_REQUEST)
     private void handleWrite(RequestMsg req, ChannelHandlerContext ctx, IServerRouter router) {
+        final int uuidTailLength = 12;
+
         LogData logData = getLogData(req.getPayload().getWriteLogRequest().getLogData());
+        Map<UUID, Long> backPointerMap = logData.getBackpointerMap();
+
+        Map<String, Long> reducedBackPointerMap = new HashMap<>(backPointerMap.size());
+        backPointerMap.forEach((uuid, data) ->{
+            String uuidStr = uuid.toString();
+            String uuidTail = uuidStr.substring(uuidStr.length() - uuidTailLength);
 
-        log.debug("handleWrite: {}, at: {}, ids: {}",
-                logData.getType(), logData.getToken(), logData.getBackpointerMap());
+            reducedBackPointerMap.put(uuidTail, data);
+        });
+
+        WRITER_LOG.debug("{}, {}, {}: {}",
+                logData.getToken().getEpoch(),
+                logData.getToken().getSequence(),
+                logData.getType(),
+                reducedBackPointerMap
+        );
```

Comment:
We only have a few thousands of streams. Even the first 8 hexadecimal digits already provides 16^8 possibilities. The chance of collision is very low. What's more important, those UUIDs are not dynamic - table names are fixed in each release and we know their UUID for sure. You can check the compactor log which has full UUIDs to have a sense if any two UUIDs have the same first 8 digits. I do agree with your second point that it's important to justify the changes by exploring the actual size saving numbers.

Reply:
I agree with George, we will never hit the issue, no reason for overprotecting oursevles agains super rare case?

Reply:
I don't consider this overprotection given that this is our go-to logs for a variety of bugs when determining if certain tables were updated during some transaction.

Can we consider using a list of pairs instead of a map for storing the truncated UUIDs and backpointers? This way, we can still reduce the number of characters printed for each UUID, but even in the presence of collisions, we will still know with certainty the number of tables that were updated in this transaction, and we will not lose any backpointer information.

Reply:
Sure, done